benchmarking/evaluations/graphing/line_graph.py

```diff
@@ -40,13 +41,36 @@ def line_graph(graph_data: LineGraphMetadata):
         (0, (3, 1, 3, 1, 1, 1, 1, 1)),  # dashdash dotdotted
     ]
     markers: List[str] = ["o", "v", "s", "d", "x", "P", "*"]
+    curr_data: GraphData
     for i, curr_data in enumerate(graph_data.data):
-        line_style = line_styles[i % len(line_styles)]
-        marker = markers[i % len(markers)]
+        # Get line styles from input, else use default values
+        line_style = (
```

Comment:
nit; 
`line_style = curr_data.line_style or line_styles[i % len(line_styles)]` seems like it does what you want - same for the instances below